### PR TITLE
Fixed Error messaging & Removed Unnecessary lines

### PR DESCRIPTION
**Pull request by achintverma**
Hello, 

While working with this wrapper I felt need of providing messages to user if something was wrong (when Paypal response carried error messages). So I added a logic to pass error message in the error object. If no error message were returned from Paypal for an error it will return generic error message (like it was already doing).

I removed a duplicate section at the bottom of the code, which was used to attach methods. 

I ran mocha tests and all tests pass.

### CODE REVIEW

File: lib/paypal-adaptive.js [hunk 1]
```diff
@@ -241,12 +249,4 @@ adaptivePaymentsMethods.forEach(function (method) {
...
 });
 
-adaptiveAccountsMethods.forEach(function (method) {
-    var prototypeMethodName = method.charAt(0).toLowerCase() + method.slice(1);
-    var apiMethodName = 'AdaptiveAccounts/' + method;
-    Paypal.prototype[prototypeMethodName] = function (data, callback) {
-        this.callApi(apiMethodName, data, callback);
-    };
-});
-
```
Comment (gaguirre) [inline lib/paypal-adaptive.js:251]: This code is to setup `AdaptiveAccounts` methods, **it is not** duplicate code

File: lib/paypal-adaptive.js [hunk 2]
```diff
@@ -133,9 +129,6 @@ Paypal.prototype.callApi = function (apiMethod, data, callback) {
     if (config.deviceIpAddress)
         options.headers['X-PAYPAL-DEVICE-IPADDRESS'] = config.deviceIpAddress;
 
-    if (config.subject)
-        options.headers['X-PAYPAL-SECURITY-SUBJECT'] = config.subject;
-
```
Comment (gaguirre) [inline lib/paypal-adaptive.js:138]: This is necessary for some people. It was added by a PR.
Why did you remove it?

**Comment by achintverma** [comment]
Oh, that was a mistake. I guess I removed it when I was removing the console logs. I'm gonna close this pull request and send another by putting those lines back.

File: lib/paypal-adaptive.js [hunk 3]
```diff
@@ -58,10 +58,7 @@ function httpsPost(options, callback) {
 
     req.on('response', function (res) {
         var response = '';
-        //do not setEndcoding with browserify
-        if (res.setEncoding) {
-          res.setEncoding('utf8');
-        }
+        res.setEncoding('utf8');
```
Comment (gaguirre) [inline lib/paypal-adaptive.js:62]: Same as below (in next comment), why did you remove this?

File: lib/paypal-adaptive.js (hunk 1 above)
Comment (achintverma) [reply lib/paypal-adaptive.js:251]: Okay, I see. Just noticed that. My bad. I'll make sure to review more carefully before sending a pull request.

Comment (gaguirre) [reply lib/paypal-adaptive.js:251]: Keep in mind you can add a commit to the same PR!
You should do that instead of create another one ;)

Comment (achintverma) [reply lib/paypal-adaptive.js:251]: Yes, I'll remember that. Thank you! and sorry again for wasting your time on review.
